sky/shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -195,17 +200,23 @@ float getDevicePixelRatio() {
     public void destroy() {
         if (mPlatformServiceProviderBinding != null) {
             mPlatformServiceProviderBinding.unbind().close();
+            mPlatformServiceProviderProxy.close();
             mPlatformServiceProvider.unbindServices();
         }
 
         if (mViewServiceProviderBinding != null) {
             mViewServiceProviderBinding.unbind().close();
+            mViewServiceProviderProxy.close();
             mViewServiceProvider.unbindServices();
         }
 
         getHolder().removeCallback(mSurfaceCallback);
         nativeDetach(mNativePlatformView);
         mNativePlatformView = 0;
+
+        mSkyEngine.close();
+        mDartServiceProvider.close();
+        mFlutterAppMessages.close();
```

Comment:
The ServiceProvider.Proxy Java objects are holding a reference to a handle.  At finalization time they think they still own the handle and will throw because the handle was never closed.

What's the right way to indicate that the handle's ownership was transferred to the engine?
Should FlutterView call serviceProviderProxy.getProxyHandler().passHandle() after marshaling the ServiceProvider over to the SkyEngine?